### DISCUSSION TITLE
ESP32 c3 Access Point problem

### DISCUSSION THREAD

**Discussion by Hichiii** [arendst/Tasmota#15908] [Q&A]
Hello,
I have a strange problem

Tasmota does an AP. I can connect from the cell phone (Android) and open the Tasmota web interface.

But I can't connect from Windows devices. When I try to connect, Windows says I can't connect.

If I load the example Access Point onto the c3 from the Arduino IDE, I can connect to it from all devices.

what is different about the Tasmota AP than the IDE example AP?

lg
Hichi

**Comment by sfromis** [comment]
What is your hardware? The Wemos C3 Mini board has bad wifi, a known troublemaker.

**Comment by Hichiii** [comment]
![Neue Bitmap](https://user-images.githubusercontent.com/107710651/176968002-85a2e156-f120-4c05-92f0-eb87cce7f151.gif)

I have an ESP32 c3 Mini 1 on a self designed board.

I added more capacitors as a test. Voltage comes from a laboratory power supply.

It always works with the Android mobile phone. If I then enter my WIFI data in the web interface, it connects to my router. Then I can also access it with the Windows laptop.

 Only the connection to the AP for the initial setup causes problems with Windows devices.

**Reply by sfromis** [reply]
The page loaded will scan for AP names to connect to. Maybe Windows throws a spanner in the works there?

**Comment by Hichiii** [comment]
I want to test that. good idea

I'm not sure in which file of Tasmota I can find wifiscan. So that I can disable it and then compile a version without Wifi Scan.

do you know that?

**Reply by sfromis** [reply]
I've not gone into details, but `xdrv_01_9_webserver.ino` does have logic for both captive portal and wifi scanning....

**Comment by Hichiii** [comment]
I tried around a bit.

I have from
    1936 int n = wifi.scannetworks ();
       the
    1936 int n = 0;
did.

Now the AP works.
I tried 10 times back and forth.
He is no longer looking for WLAN.
It no longer shows WLAN in the web interface.
The button Scan for Networks is still there, but nothing happens when you click on it.

When I enter my Wland SSID I can connect and everything works.
Now we already know that it is probably related to it.
But I don't know why the WiFiscan with Android does not cause any problems and only with Windows.

Would be nice if I could somehow solve that it works properly. Am I the only one with the problem or is it a general problem with the C3?

**Comment by Hichiii** [comment]
In the debug I see that the ESP32 c3 needs 5 seconds for the search.

I did the same test with an ESP8266, it only takes 2 seconds.


I thought it was about time.

I then installed a delay(5000) instead of WiFi.scanNetworks() to test it.

But that doesn't bother me, the AP and the web interface still work.


But as soon as WiFi.scanNetworks() is in it, it is not possible to connect to the AP on the ESP32 c3 mini 1 with Windows.



Does anyone have an idea? It would be nice if there was a way to get it to work properly


lg

Hichi

**Reply by sfromis** [reply]
What I'd expect, based on seeing that Wifi scan in other circumstances, would be that it would be "dead" on the regular wifi while scanning, but getting back after the scan completed. If the scan could not complete for whatever reason, trouble.....
One quite relevant debugging step would be to follow it in the serial console while things are going on.

**Comment by Hichiii** [comment]
Here is the output from the log:

00:00:00.000 HDW: ESP32-C3 <\r><\n>
00:00:00.007 UFS: FlashFS mounted with 308 kB free<\r><\n>
00:00:00.021 CFG: Loaded from File, Count 5<\r><\n>
00:00:00.026 QPC: Count 1<\r><\n>
00:00:00.027 CFG: CR 379/699, Busy 0<\r><\n>
00:00:00.029 RED: Mode 1<\r><\n>
00:00:00.058 CFG: No '*.autoconf' file found<\r><\n>
00:00:00.062 BRY: Berry initialized, RAM used=4203 bytes<\r><\n>
00:00:00.068 BRY: No 'preinit.be'<\r><\n>
00:00:00.070 SRC: Restart<\r><\n>
00:00:00.072 Project tasmota - Tasmota Version 12.0.2(tasmota)-2_0_3(2022-07-02T15:50:36)<\r><\n>
00:00:00.073 UFILESYSTEM OK!<\r><\n>
00:00:00.092 BRY: No 'autoexec.be'<\r><\n>
00:00:00.230 WIF: WifiManager active for 3 minutes<\r><\n>
00:00:00.458 WIF: Wifimanager set AccessPoint<\r><\n>
00:00:00.967 HTP: Web server active on tasmota-66C4B4-1204 with IP address 192.168.4.1<\r><\n>
00:00:06.687 QPC: Reset<\r><\n>
00:00:08.689 APP: Boot Count 3<\r><\n>
00:00:09.003 CFG: Saved, Count 6, Bytes 4096<\r><\n>
00:00:20.864 HTP: Redirected to captive portal<\r><\n>
00:00:20.914 HTP: Redirected to captive portal<\r><\n>
00:00:21.014 HTP: Blank Device - Initial Configuration<\r><\n>
00:00:21.014 HTP: Configure WiFi<\r><\n>
00:00:21.039 WIF: Scanning...<\r><\n>
00:00:25.941 WIF: Scan done<\r><\n>

When I try to connect Windows to the AP it says Scanning, Scan done.
Then nothing happens anymore.

When I connect to Android it looks like this:

licensed, RAM used=4203 bytes<\r><\n>
00:00:00.070 BRY: No 'preinit.be'<\r><\n>
00:00:00.072 SRC: Restart<\r><\n>
00:00:00.074 Project tasmota - Tasmota Version 12.0.2(tasmota)-2_0_3(2022-07-02T15:50:36)<\r><\n>
00:00:00.075 UFILESYSTEM OK!<\r><\n>
00:00:00.094 BRY: No 'autoexec.be'<\r><\n>
00:00:00.232 WIF: WifiManager active for 3 minutes<\r><\n>
00:00:00.460 WIF: Wifimanager set AccessPoint<\r><\n>
00:00:00.969 HTP: Web server active on tasmota-66C4B4-1204 with IP address 192.168.4.1<\r><\n>
00:00:04.730 HTP: Redirected to captive portal<\r><\n>
00:00:04.832 HTP: Blank Device - Initial Configuration<\r><\n>
00:00:04.832 HTP: Configure WiFi<\r><\n>
00:00:04.863 WIF: Scanning...<\r><\n>
00:00:09.769 WIF: Scan done<\r><\n>
00:00:12.426 QPC: Reset<\r><\n>
00:00:13.919 APP: Boot Count 4<\r><\n>
00:00:14.343 CFG: Saved, Count 7, Bytes 4096<\r><\n>
00:00:14.871 HTP: Redirected to captive portal<\r><\n>
00:00:29.993 HTP: Redirected to captive portal<\r><\n>
00:00:30.093 HTP: Redirected to captive portal<\r><\

**Reply by sfromis** [reply]
Since it does get to `00:00:25.941 WIF: Scan done`, but not to `QPC: Reset`, it looks like it stalled badly, as the latter message should occur after every boot, to indicate that it got to the point of the boot counting as successful. Maybe the logic to resume normal Wifi after the scanning breaks on ESP32-C3?

**Comment by Hichiii** [comment]
yes something like that it can be.  but it happens to me when I connect to windows. With all Windows laptop.  i don't know how i could change the code to do more tests. do you have any idea what else i could test?LgHichiAm 03.07.22, 00:01 schrieb sfromis ***@***.***>:

   
  Since it does get to 00:00:25.941 WIF: Scan done, but not to QPC: Reset, it looks like it stalled badly, as the latter message should occur after every boot, to indicate that it got to the point of the boot counting as successful. Maybe the logic to resume normal Wifi after the scanning breaks on ESP32-C3?

**Reply by sfromis** [reply]
The logical thing to look for would be differences in network traffic depending on which OS is being used.

**Comment by Hichiii** [comment]
I tried to find out something with wireshark.

I think the C3 can't have the AP connection at the same time as the WiFi.scanNetworks().

I have no idea how to solve the problem in Tasmota.

I added WiFi.scanNetworks() to the Arduino AP Sketch. Then the same problem is there.

It doesn't bother the ESP8266. only the ESP32 c3.

If anyone has an idea how to use Tasmota for the ESP32 c3 I would be very happy.

**Reply by sfromis** [reply]
I never had any trouble with ESP32-C3, but now that seems to be due to my general workflow of just not using the wifi onboarding, but instead doing basic config via commands over serial, right after flashing. While the wifi onboarding may be "nice to have" during initial flashing, that's also the time where I already have wires connected. Later, in a closed-off device, it becomes more of a stumbling block to not have this avenue.

**Reply by sfromis** [reply]
BTW. When using Wireshark, did you spot any "interesting" difference when using Windows vs Android?

**Reply by barbudor** [reply]
I don't think Wireshark would show anything until Wifi is connected

Anyhow, I pickup a C3 board, used web installer to erase and flash tasmota32c3
I had no problems to connect to hte hot-spot from Tasmota
Although scanning seems to take longer on C3 than on ESP8266 (from what I recall, I didn't checked)

Serial logs after a `backlog seriallog 4; restart 1`
```
00:00:00.001 HDW: ESP32-C3 
00:00:00.007 UFS: FlashFS mounted with 308 kB free
00:00:00.017 CFG: Loaded from File, Count 10
00:00:00.021 QPC: Count 1
00:00:00.022 CFG: CR 379/699, Busy 0
00:00:00.024 ROT: Mode 1
00:00:00.029 BRY: GC from 3586 to 3386 bytes, objects freed 4/19 (in 1 ms) - slots from 31/61 to 23/61
00:00:00.052 CFG: No '*.autoconf' file found
00:00:00.055 BRY: GC from 5286 to 4203 bytes, objects freed 6/38 (in 1 ms) - slots from 56/76 to 35/76
00:00:00.056 BRY: Berry initialized, RAM used=4203 bytes
00:00:00.072 BRY: No 'preinit.be'
00:00:00.073 SRC: Restart
00:00:00.076 Project tasmota - Tasmota Version 12.0.2(tasmota)-2_0_3(2022-06-20T12:41:30)
00:00:00.093 BRY: No 'autoexec.be'
00:00:00.230 WIF: Checking connection...
00:00:00.230 WIF: WifiManager active for 3 minutes
00:00:00.461 WIF: Wifimanager set AccessPoint
00:00:00.970 HTP: Web server active on tasmota-3B65B0-1456 with IP address 192.168.4.1
00:00:06.946 QPC: Reset
00:00:08.949 APP: Boot Count 6
00:00:09.986 CFG: Saved, Count 11, Bytes 4096
00:00:12.780 HTP: Redirected to captive portal
00:00:13.130 HTP: Redirected to captive portal
00:00:17.293 HTP: Blank Device - Initial Configuration
00:00:17.293 HTP: Configure WiFi
00:00:17.323 WIF: Scanning...
00:00:22.627 WIF: Scan done
00:00:27.446 HTP: Redirected to captive portal
00:00:27.744 HTP: Redirected to captive portal
```

![image](https://user-images.githubusercontent.com/2996042/177211715-0475848c-e362-405d-9793-8690669e85e4.png)

**Comment by Hichiii** [comment]
![WhatsApp Image 2022-07-05 at 12 21 17](https://user-images.githubusercontent.com/107710651/177307215-7f80b327-40e5-4cf0-989b-0543065a1bc6.jpeg)

Do you also use a c3 mini 1?

Maybe there is a difference in the hardware to the other c3?

I have 30 of these here. Same problem with everything

I just found this:
https://github.com/espressif/arduino-esp32/issues/4554
This seems to be the same problem.

**Reply by barbudor** [reply]
no, the module on my board is an AI Thinker ESP-C3-32S

The linked issue seems to be for ESP32 standard, not C3. Can't tell if this is the same or not.
However, I can' recall anyone has reported this problem wiht Tasmota and ESP32 standard which would IMHO exclude this Issue report.

**Reply by sfromis** [reply]
Did a test with an early "sample" Espressif ESP32-C3-DEVKITC-02v1.1 (ESP32-C3 rev.2), and a regular production ESP-C3-12F module (ESP32-C3 rev.3), current dev `tasmota32c3.factory.bin` (flashed using esptool.py), which confirms that there is a problem attempting to onboard via Windows 10, newish laptop with regular software updates. Captive portal worked from an old Android phone.

Having a serial console connected during the process, output matches what Hitchiii is seeing:
```
...
00:00:00.084 Project tasmota - Tasmota Version 12.0.2.2(tasmota)-2_0_3(2022-07-05T20:21:11)
00:00:00.096 BRY: Error be_top is non zero=1
00:00:00.247 WIF: Checking connection...
00:00:00.248 WIF: WifiManager active for 3 minutes
00:00:00.476 WIF: Wifimanager set AccessPoint
00:00:00.982 HTP: Web server active on tasmota-760520-1312 with IP address 192.168.4.1
00:00:06.687 QPC: Reset
00:00:08.689 APP: Boot Count 7
00:00:09.055 CFG: Saved, Count 11, Bytes 4096
```
... connecting from Windows at this point, messages appearing looks normal
```
00:00:34.362 HTP: Redirected to captive portal
00:00:34.662 HTP: Redirected to captive portal
00:00:35.218 HTP: Redirected to captive portal
00:00:37.123 HTP: Blank Device - Initial Configuration
00:00:37.123 HTP: Configure WiFi
00:00:37.151 WIF: Scanning...
00:00:42.253 WIF: Scan done
status
```
At this point Tasmota is frozen, does not react to a command from the serial console. Connecting from the Android device, I see the same messages, but 2.6 seconds later (after the browser was started on the phone), I get `HTP: Redirected to captive portal` again, with the regular onboarding page appearing.

The alternative flashing/onboarding via Tasmota Web Installer (meaning not using Windows 10 to manually connect to the Tasmota onboarding AP, and not using the captive portal) worked, but of course that's a different procedure.

Of course, no problem doing the onboarding via commands over serial. You should be able to onboard this way, or via the Tasmota web installer, but obviously the captive portal onboarding should work too.

**Reply by barbudor** [reply]
Personnaly I got rid of Wifi on-boarding by compiling my own binaries with Wifi SSID and Password hardcoded in user_config_override.h
I fear no unexpected power-cycle device recovery 😂

**Comment by Hichiii** [comment]
Hi,
Yes, I can also log into my wifi directly while compiling.
This way you can avoid the problem

Of course, it would be nice if this would work completely. also the captive portal.
probably there is no way with the .esp32 c3 mini 1

lg
Hichi

**Reply by sfromis** [reply]
Arguably, it is a bug, which "someone" could spend time on analyzing and fixing.

Anyway, since you use serial for flashing, you need no build time config or captive portal for initial setup, you can just use serial commands. Me using that method is an additional reason why I never had the Windows captive portal problem (besides not normally using Windows).
